src/librustc_mir/transform/uninhabited_enum_branching.rs

```diff
@@ -0,0 +1,129 @@
+//! A pass that eliminates branches on uninhabited enum variants.
+
+use rustc::mir::{
+    BasicBlock, Body, Local, Operand, Place, PlaceBase, Rvalue, StatementKind, TerminatorKind
+};
+use rustc::ty::{Ty, TyCtxt};
+use rustc::ty::layout::{Abi, TyLayout, Variants};
+use crate::transform::{MirPass, MirSource};
+
+pub struct UninhabitedEnumBranching;
+
+fn get_discriminant_local(terminator: &TerminatorKind<'_>) -> Option<Local> {
+    if let TerminatorKind::SwitchInt { discr: Operand::Move(Place { base: PlaceBase::Local(l),
+                                                                    projection: box [] }),
+                                       .. } = terminator {
```

Comment:
what's with the formating? (might as well run the whole file through rustfmt since it's new).